src/libs/actions/Report.js

```diff
@@ -544,16 +546,23 @@ function openReport(reportID, participantList = [], newReportObject = {}) {
         onyxData.optimisticData[0].onyxMethod = CONST.ONYX.METHOD.SET;
 
         // Also create a report action so that the page isn't endlessly loading
+        const optimisticReportAction = ReportUtils.buildOptimisticCreatedReportAction(newReportObject.ownerEmail);
+        optimisticReportActionID = NumberUtils.rand64();
+        // optimisticReportAction[optimisticReportActionID] = optimisticReportAction[0];
+        // delete optimisticReportAction[0];
+        // console.log('sending optimisticReportActionID', optimisticReportActionID);
+        optimisticReportAction[0].reportActionID = optimisticReportActionID;
         onyxData.optimisticData[1] = {
             onyxMethod: CONST.ONYX.METHOD.SET,
             key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
-            value: ReportUtils.buildOptimisticCreatedReportAction(newReportObject.ownerEmail),
+            value: optimisticReportAction,
         };
     }
     API.write('OpenReport',
         {
             reportID,
             emailList: participantList ? participantList.join(',') : '',
+            createdReportActionID: optimisticReportActionID,
```

Comment:
Following the earlier suggestion, can be replaced with `optimisticReportAction. reportActionID` instead of having `optimisticReportActionID` here.